commands/root.go

```diff
@@ -40,7 +41,17 @@ func NewRootCmd(name string, isPlugin bool, dockerCli command.Cli) *cobra.Comman
 			}
 			cmd.SetContext(appcontext.Context())
 			if !isPlugin {
-				return nil
+				// InstallFlags and SetDefaultOptions are necessary to match
+				// the plugin mode behavior to handle env vars such as
+				// DOCKER_TLS, DOCKER_TLS_VERIFY, ... and we also need to use a
+				// new flagset to avoid conflict with the global debug flag
+				// that we already handle in the root command otherwise it
+				// would panic.
+				nflags := pflag.NewFlagSet(cmd.DisplayName(), pflag.ContinueOnError)
+				options := cliflags.NewClientOptions()
+				options.InstallFlags(nflags)
```

Comment:
I'm confused what is docker-specific in this `InstallFlags` call? And if there is nothing docker-specific and just needs to be called then why isn't `NewClientOptions()` handling it automatically.

Reply:
> Or is the difference from calling this function in `PersistentPreRunE` instead of before `NewRootCmd()`?

No there isn't bu I was hoping to be able to use `options.InstallFlags(cmd.Flags())` here but it panics because we already declare a global `debug` flag.